var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -125,10 +125,20 @@ class Boost(Package):
     depends_on('mpi', when='+mpi')
     depends_on('bzip2', when='+iostreams')
     depends_on('zlib', when='+iostreams')
+    depends_on('zlib')
```

Comment:
Somehow, this conditional dependencies was making problem, so I tested but forgot to remove it.